modules/ting_subsearch/ting_subsearch_secondary/js/subsearch_secondary.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * @file
+ * Secondary search JS functionality.
+ */
+
+(function ($) {
+  'use strict';
+
+  var settings = Drupal.settings.tingSubsearchSecondary;
+
+  $.ajax({
+    type: 'POST',
+    url: '/subsearch_secondary',
+    data: {
+      'keys': settings.keys,
+      'numResults': settings.numResults
+    },
+  }).done(function (r) {
+    if (r !== '') {
+      var message = $(r);
+      $('#subsearch-secondary').html(message);
+      Drupal.attachBehaviors(message);
+    }
+  }).fail(function (e) {
+    console.log(e);
```

Comment:
An option would be just fail silently. I see this approach in other places.

If something with the search or suggestion went wrong it will be logged on server.

Reply:
Removed in new implementation of ajax subsearches.